src/ort_genai_c.h

```diff
@@ -170,38 +176,55 @@ OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsTryGraphCaptureWithMaxBatch
  */
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputSequences(OgaGeneratorParams* generator_params, const OgaSequences* sequences);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputSequences(OgaGeneratorParams* generator_params,
+                                                                       const OgaSequences* sequences);
 
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputs(OgaGeneratorParams* generator_params, const OgaNamedTensors* named_tensors);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetInputs(OgaGeneratorParams* generator_params,
+                                                               const OgaNamedTensors* named_tensors);
 
 /*
- * \brief For additional model inputs that genai does not handle, this lets the user set their values. For example LoRA models handle
- * fine tuning through model inputs. This lets the user supply the fine tuning inputs, while genai handles the standard inputs.
- * \param[in] generator_params The generator params to set the input on
- * \param[in] name Name of the model input (this must match the model's input name)
- * \param[in] tensor The OgaTensor of the input data
+ * \brief For additional model inputs that genai does not handle, this lets the user set their values. For example LoRA
+ * models handle fine tuning through model inputs. This lets the user supply the fine tuning inputs, while genai handles
+ * the standard inputs. \param[in] generator_params The generator params to set the input on \param[in] name Name of the
+ * model input (this must match the model's input name) \param[in] tensor The OgaTensor of the input data
  */
-OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetModelInput(OgaGeneratorParams* generator_params, const char* name, OgaTensor* tensor);
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetModelInput(OgaGeneratorParams* generator_params,
+                                                                   const char* name, OgaTensor* tensor);
 
 OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetWhisperInputFeatures(OgaGeneratorParams*, OgaTensor* tensor);
 
+/// <summary>
+/// Set active adapter names for the give params instance. The specified adapter names
+/// replace any previously set names.
+/// This call has no effect on any Generator instances created before this call is made.
+/// </summary>
+/// <param name="">a valid params ptr</param>
+/// <param name="adapters">an array of valid adapter names (UTF-8 encoded C strings). This array replaces
+///                        any previously set active adapters within GeneratorParams.</param>
+/// <param name="count">size of the names array. Passing zero removes any previously set adapters.</param>
+/// <returns>Returns an error if a non-existent adapter name is submitted</returns>
+OGA_EXPORT OgaResult* OGA_API_CALL OgaGeneratorParamsSetActiveAdapters(OgaGeneratorParams* params,
```

Comment:
Caching and memory management will be addressed in a separate PR as mentioned in the PR description. 
This would be mostly related to memory management on the device, we can also introduce memmap for the files.

Reply:
@baijumeswani if needed APIs to add lora params piecemeal can be added later. It should be an incremental change once this infrastructure to manage the adapters is in.